src/terminfo/parser/compiled.rs

```diff
@@ -157,8 +158,22 @@ pub static stringnames: &'static[&'static str] = &[ "cbt", "_", "cr", "csr", "tb
     "OTG3", "OTG1", "OTG4", "OTGR", "OTGL", "OTGU", "OTGD", "OTGH", "OTGV", "OTGC", "meml", "memu",
     "box1"];
 
+fn read_harder<R: io::Read>(reader: &mut R, buffer: &mut [u8]) -> io::Result<usize> {
```

Comment:
This may be better named `read_all`, and I think it may be better structured as returning `io::Result<()>` and returning an error on EOF. I also wouldn't bother handling `ErrorKind::Interrupted` here just yet, that should be punted upwards wherever possible.